### ISSUE TITLE
Support for ExternalIPs in Kubernetes Service

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#5293]
```
        // externalIPs is a list of IP addresses for which nodes in the cluster
        // will also accept traffic for this service.  These IPs are not managed by
        // Kubernetes.  The user is responsible for ensuring that traffic arrives
        // at a node with this IP.  A common example is external load-balancers
        // that are not part of the Kubernetes system.
        // +optional
        ExternalIPs []string `json:"externalIPs,omitempty" protobuf:"bytes,5,rep,name=externalIPs"`
```

cc @kbaegis

**Comment by kbaegis** [comment]
Particular use-case:
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    app: nulllabs-vpn
  name: nulllabs-vpn
spec:
  replicas: 2
  minReadySeconds: 20
  selector:
    matchLabels:
      app: nulllabs-vpn
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: nulllabs-vpn
    spec:
      dnsConfig:
        nameservers:
        - 172.21.255.101
        - 172.23.255.254
        options:
        searches:
        - mgmt
        - home
        - lab
        - vpn
      restartPolicy: Always
      hostAliases:
      - ip: "127.0.0.1"
        hostnames:
        - "router"
      - ip: "172.21.0.1"
        hostnames:
        - "router.mgmt"
      - ip: "172.21.255.254"
        hostnames:
        - "services.mgmt"
      hostNetwork: true
      containers:
      - image: crucible.lab:4000/oci/strongswan:2018-08-19
        imagePullPolicy: Always
        name: strongswan
        ports:
        - containerPort: 500
          protocol: UDP
        - containerPort: 4500
          protocol: UDP
        - containerPort: 1701
          protocol: UDP
        volumeMounts:
        securityContext:
          privileged: true
          capabilities:
            add:
              - NET_ADMIN
              - NET_RAW
      nodeSelector:
        type: router
status: {}
---
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: null
  labels:
    app: nulllabs-vpn
  name: nulllabs-vpn
spec:
  externalIPs:
  - 192.168.0.254
  - 192.168.0.253
  - 192.168.0.252
  ports:
  - port: 500
    name: vpn-500
    protocol: UDP
    targetPort: 500
  - port: 4500
    name: vpn-4500
    protocol: UDP
    targetPort: 4500
  - port: 1701
    name: vpn-1701
    protocol: UDP
    targetPort: 1701
  sessionAffinity: ClientIP
  selector:
    app: nulllabs-vpn
status:
  loadBalancer: {}
```
![public network](https://user-images.githubusercontent.com/306920/44867362-dbc9f700-ac44-11e8-9d98-81e623a3926b.png)

**Comment by Kami-no** [comment]
It perfectly works with [MetalLB](https://metallb.universe.tf/tutorial/layer2/)

**Comment by kbaegis** [comment]
This appears to be working. My usecase was hosting DNS inside k8s and managing a VIP with keepalived for survivability while allowing the hosts to pull from the internal k8s dns service.

**Comment by aanm** [comment]
@antonmry @jcleira @kbaegis This is going to be implemented by #8455
